feat: move hmac fns to own package

bitgoAaron opened the pull request:
- define hmac related functions in separate module

Ticket: CE-5010

<!--
# Please be aware of the following when making your pull request:

## Description

Please include a summary of your proposed changes and which issue is being addressed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Issue Number

Internal Users - Please include the related internal tracking number (e.g. BG-000000).
External Users - Please link to any relevant github issues as necessary.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code compiles correctly for both Node and Browser environments
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and I have properly described any BREAKING CHANGES
- [ ] The ticket or github issue was included in the commit message as a reference
- [ ] I have made corresponding changes to the documentation and on any new/updated functions and/or methods - [jsdoc](https://jsdoc.app/)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
-->

bitgoAaron committed:
feat: move hmac fns to own package

- define hmac related functions in
  separate module

Ticket: CE-5010

alebusse commented:
does it need to be added to the dockerfile https://github.com/BitGo/BitGoJS/blob/master/Dockerfile?

zahin-mohammad commented on modules/sdk-hmac/README.md:
````diff
@@ -0,0 +1,20 @@
+
+## Installation
+
+```shell
+npm i @bitgo/sdk-api @bitgo/sdk-lib-mpc @bitgo/sdk-opensslbytes
+```
+
+Import the `openSSLBytes` from this module & pass to related functions that expect it:
+
+```javascript
+import { loadWebAssembly } from '@bitgo/sdk-opensslbytes';
+import { EcdsaRangeProof } from '@bitgo-beta/sdk-lib-mpc';
+
+const openSSLBytes = loadWebAssembly().buffer;
+
+const nTilde = await EcdsaRangeProof.generateNtilde(openSSLBytes, 3072);
+```
````
Copy pasta? Also why is there already a changelog for this module?

zahin-mohammad commented on modules/sdk-hmac/package.json:
```diff
@@ -0,0 +1,49 @@
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/BitGo/BitGoJS.git",
+    "directory": "modules/sdk-opensslbytes"
```
Needs an update.

zahin-mohammad commented on modules/sdk-hmac/CHANGELOG.md:
```diff
@@ -0,0 +1,19 @@
+
+# 2.0.0 (2024-08-20)
+
+### Features
+
+- move opensslbytes to own package ([e23c562](https://github.com/BitGo/BitGoJS/commit/e23c5627957916055e68329541dd1eb775704ca5))
+
+### BREAKING CHANGES
+
+- clients using challenge
+  generation & TSS Recovery functions must now
+  install @bitgo/sdk-opensslbytes separately &
+  provide the openSSLBytes WASM themselves.
+
+Ticket: CE-4329
```
Needs an update.

zahin-mohammad commented on modules/sdk-hmac/.eslintignore:
```diff
@@ -0,0 +1,2 @@
+dist/
+src/opensslbytes.ts
```
Needs an update.

bitgoAaron committed:
fix: remove copy/pasta from sdk-hmac module

- update Dockerfile
- update copied contents to reflect current
  module

Ticket: CE-5010
